manager/src/test/java/com/redhat/service/smartevents/manager/api/user/validators/processors/ProcessorGatewayConstraintValidatorTest.java

```diff
@@ -207,36 +196,26 @@ void isValid_sourceWithTransformationIsNotValid(Gateway gateway) {
 
         if (p.getType() == ProcessorType.SOURCE) {
             assertThat(constraintValidator.isValid(p, validatorContextMock)).isFalse();
-            verifyGetValidatorCall(gateway, never(), ArgumentMatchers::anyString);
-            verifyIsValidCall(gateway, never());
+            verifyGetValidatorCall(never(), ArgumentMatchers::anyString);
+            verifyIsValidCall(never());
             verifyErrorMessage(SOURCE_PROCESSOR_WITH_TRANSFORMATION_ERROR);
         } else {
             assertThat(constraintValidator.isValid(p, validatorContextMock)).isTrue();
-            verifyGetValidatorCall(gateway, times(1), gateway.getType());
-            verifyIsValidCall(gateway, times(1));
+            verifyGetValidatorCall(times(1), gateway.getType());
+            verifyIsValidCall(times(1));
         }
     }
 
-    private void verifyGetValidatorCall(Gateway gateway, VerificationMode mode, String argument) {
-        verifyGetValidatorCall(gateway, mode, () -> argument);
+    private void verifyGetValidatorCall(VerificationMode mode, String argument) {
+        verifyGetValidatorCall(mode, () -> argument);
     }
 
-    private void verifyGetValidatorCall(Gateway gateway, VerificationMode mode, Supplier<String> argumentSupplier) {
-        if (gateway instanceof Action) {
-            verify(gatewayConfiguratorMock, mode).getActionValidator(argumentSupplier.get());
-        } else {
-            verify(gatewayConfiguratorMock, mode).getSourceValidator(argumentSupplier.get());
-        }
+    private void verifyGetValidatorCall(VerificationMode mode, Supplier<String> argumentSupplier) {
```

Comment:
Lots of mocking in this class... what are we even testing? I'm not sure we need this anymore

Reply:
This is checking that the top level `ProcessorGatewayConstraintValidator` is actually calling the Validator. I'd suggest to review this in another PR